File: backend/school-app-backend/Features/Students/Student.cs
using System.ComponentModel.DataAnnotations;

namespace school_app_backend.Features.Students;

public class Student
{
    public int Id { get; set; }
    public required string FirstName { get; set; }
    public string? MiddleName { get; set; }
    public required string LastName { get; set; }
    public string? SecondLastName { get; set; }
    public required string FullName { get; set; }
    public string? Address { get; set; }
    
    [AllowedValues("M", "F")] 
    public required char Gender { get; set; }
    public required DateTime DateOfBirth { get; set; }
    public int Age { get; set; }    
    [EmailAddress] 
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public required DateTime CreationDate { get; set; }
    public required DateTime LastUpdateDate { get; set; }
    public required int TutorQuantity { get; set; }
    public required int ActualGrade { get; set; }
    public required bool IsDeleted { get; set; } = false;
    public required bool IsActive { get; set; }
}

File: school-app-mobile/models/students/Student.ts
export interface Student {
  id: number;
  fullName: string;
  firstName: string;
  middleName?: string;
  lastName: string;
  secondLastName?: string;
  gender: "M" | "F";
  dateOfBirth: Date;
  actualGrade: string;
  pictureUrl: string;
}

export type StudentForTutor = Pick<Student, "id" | "fullName" | "actualGrade">;

export interface StudentApiResponseForTutor {
  data: StudentForTutor[];
}
export type StudentApiResponseForTeacher = Pick<Student, "fullName">;
